src/lightning_fabric/strategies/deepspeed.py

```diff
@@ -365,24 +366,123 @@ def module_sharded_context(self) -> Generator[None, None, None]:
     def save_checkpoint(
         self, path: _PATH, state: Dict[str, Union[Module, Optimizer, Any]], storage_options: Optional[Any] = None
     ) -> None:
-        raise NotImplementedError
+        """Save model, optimizer, and other state in a checkpoint directory.
+
+        Args:
+            path: A path to where the files should be saved
+            state: A dictionary with contents to be saved. If the dict contains modules or optimizers, their
+                state-dict will be retrieved and converted automatically.
+            storage_options: Unused by this strategy, since it doesn't use a ``CheckpointIO`` plugin.
```

Comment:
I was looking at the other implementations. Where do we even use `storage_options`? Why did we add them?

I couldn't find it, maybe we should remove it

Reply:
PyTorch Lightning imports these checkpoint io implementations, and storage options is used there from trainer.save_checkpoint() and passed down. Feature request by Meta

Reply:
Right but none of our CheckpointIO implementations support it. I'd advocate for removal (in a separate PR). Do you know where this was discussed originally?